app/src/state/wallet.tsx

```diff
@@ -0,0 +1,154 @@
+    case 'updateSelectedNetwork':
+      return { ...state, selectedNetwork: action.payload }
+    case 'updateSelectedAccount':
+      return { ...state, selectedAccount: action.payload }
+    default:
+      return state
+  }
+}
+
+interface WalletStateProviderInterface {
+  children?: React.ReactNode
+}
+
+// Wallet State Provider
+const WalletStateProvider = (props: WalletStateProviderInterface) => {
+  const { children } = props
+
+  // Swap Methods
+  const {
+    getAllTokens,
+    getBalance,
+    getERC20TokenBalance,
+    getSelectedNetwork,
+    getSelectedAccount
+  } = useSwapContext()
+
+  // Wallet State
+  const [state, dispatch] = useReducer(WalletReducer, initialState)
+  const { tokenList, selectedAccount, selectedNetwork, tokenBalances } = state
+
+  let didInit = false
+
+  if (!didInit) {
+    didInit = true
+    // Gets Selected Network and then sets to state
+    if (selectedNetwork.chainId === undefined) {
+      getSelectedNetwork()
+        .then((result) =>
+          dispatch({ type: 'updateSelectedNetwork', payload: result })
+        )
+        .catch((error) => console.log(error))
+    }
+
+    // Gets Selected Account and then sets to state
+    if (selectedAccount === '') {
+      getSelectedAccount()
+        .then((result) =>
+          dispatch({ type: 'updateSelectedAccount', payload: result })
+        )
+        .catch((error) => console.log(error))
+    }
+
+    // Gets all tokens and then sets to state
+    if (tokenList.length === 0) {
+      getAllTokens(selectedNetwork.chainId, selectedNetwork.coin)
+        .then((result) =>
+          dispatch({ type: 'updateTokenList', payload: result.tokens })
+        )
+        .catch((error) => console.log(error))
+    }
+
+    // Gets all balances and sets to state
+    if (!tokenBalances[NATIVE_ASSET_CONTRACT_ADDRESS_0X]) {
+      let balances = tokenBalances
+      Promise.all(
+        tokenList.map(async (token) => {
+          if (token.contractAddress === NATIVE_ASSET_CONTRACT_ADDRESS_0X) {
+            // Get Native Token Balance
+            await getBalance(
+              selectedAccount,
+              selectedNetwork.coin,
+              selectedNetwork.chainId
+            )
+              .then((result) => {
+                balances[NATIVE_ASSET_CONTRACT_ADDRESS_0X] = result.balance
+                dispatch({ type: 'updateTokenBalances', payload: balances })
+              })
+              .catch((error) => console.log(error))
+            return
+          }
+          // Get ERC721 Token Balances
+          await getERC20TokenBalance(
+            token.contractAddress,
+            selectedAccount,
+            token.chainId
+          )
+            .then((result) => {
+              balances[token.contractAddress] = result.balance
+              dispatch({ type: 'updateTokenBalances', payload: balances })
+            })
+            .catch((error) => console.log(error))
+        })
+      )
+    }
+  }
```

Comment:
Data fetching should be conducted via a useEffect with a cleanup to prevent state updates to unmounted components
https://beta.reactjs.org/learn/you-might-not-need-an-effect#fetching-data
https://beta.reactjs.org/learn/you-might-not-need-an-effect#initializing-the-application

Reply:
Moved into a useEffect Hook!